include/crocoddyl/core/actions/lqr.hxx

```diff
@@ -38,6 +38,7 @@ void ActionModelLQRTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstract>
     throw_pretty("Invalid argument: "
                  << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
   }
+  boost::shared_ptr<ActionDataLQRTpl<Scalar>> d = boost::static_pointer_cast<ActionDataLQRTpl<Scalar>>(data);
```

Comment:
I didn't realize this before.

We should cast a raw pointer, instead. This avoids to create an owner.
It should be something like this:
```c++
 ​  Data* d = ​static_cast​<Data*>(data.​get​());
```